### ISSUE TITLE
Intense flashing when an item frame is in field of view

### ISSUE THREAD

**Issue by Y4k3d0** [ComplementaryDevelopment/ComplementaryReimagined#169]
When looking directly at an item frame, nothing significant happens, yet it sometimes flashes the screen with a very bright light when looked at in very specific angles. As such, this bug isn't easy to reproduce and is quite unpredictable.

A very subtle change in angle can make it disappear suddenly, which may cause it to flash rapidly and cause discomfort for the player.

I haven't tried it on its glowing counterpart, but here are pictures showing what can be seen when it occurs :
![2024-12-21_00 32 40](https://github.com/user-attachments/assets/90ed0286-80e6-478e-8dba-2be3192fe1ed)
![2024-12-21_00 32 38](https://github.com/user-attachments/assets/45dea76c-5b2d-463b-ab15-a8789b0bab94)

Here's what the scene is supposed to look like :
![2024-12-24_13 00 19](https://github.com/user-attachments/assets/083e0561-55ae-4dad-8ea7-e1a396bfef24)

**Comment by EminGT** [comment]
We have never seen a problem like this before. Do you have any extra mods installed?

**Comment by Y4k3d0** [comment]
Yes, I am running on a heavily modded version. Here's my mod list :
```
- appleskin 3.0.6+mc1.21
- architectury 13.0.8
- badoptimizations 2.2.1
- betterf3 11.0.3
- cloth-config 15.0.140
- coroutil 1.21.0-1.3.7
- cubes-without-borders 3.0.0+mc1.21
- debugify 1.21.1+1.0
- dynamic_fps 3.7.7
- dynamiccrosshair 9.3
- effecttimerplus 2.2.6+1.21
- emi 1.1.18+1.21.1+fabric
- emojitype 2.3.0-1.21.3
- enhancedblockentities 0.10.2+1.21
- fabric-api 0.110.0+1.21.1
- fabricloader 0.16.9
- fastquit 3.0.0+1.20.6
- flashback 0.15.0
- forgeconfigapiport 21.1.3
- heywiki 1.7.2+1.21.1
- highlighter 1.1.11
- iceberg 1.2.7
- immediatelyfast 1.3.3+1.21.1
- iris 1.8.1+mc1.21.1
- java 21
- languagereload 1.7.2+1.21.1
- litematica 0.19.53
- make_bubbles_pop 0.3.0-fabric
- malilib 0.21.4
- mcmouser 1.21-fabric-1.0.0
- minecraft 1.21.1
- minihud 0.32.53
- modelfix 1.21-1.6
- modmenu 11.0.3
- moreculling 1.0.2
- mru 1.0.7+1.21+fabric
- notenoughanimations 1.8.2
- owo 0.12.15+1.21
- particlerain 3.0.2
- particular 1.1.1+1.21
- pickupnotifier 21.1.0
- puzzleslib 21.1.22
- reeses-sodium-options 1.8.0+mc1.21.3
- screenshot_viewer 1.3.3-fabric-mc1.21
- shouldersurfing 4.6.4
- smoothscroll 2.2.1
- sodium 0.6.5+mc1.21.1
- sodium-extra 0.6.0+mc1.21.1
- sound_physics_remastered 1.21.1-1.4.8
- tia 1.2.1
- voicechat 1.21.1-2.5.27
- watut 1.21.0-1.1.3
- yet_another_config_lib_v3 3.6.2+1.21-fabric
```

Here's also my hardware information :
CPU: 8x Apple M1
GPU: Apple M1 (Supports OpenGL 4.1 Metal - 89.3)
OS: Mac OS X (15.0.1)

I'm currently using the Potato profile with minor tweaks, here's what the config file looks like :
```
#Tue Dec 24 14:27:29 CET 2024
AURORA_STYLE_DEFINE=2
BLOCK_REFLECT_QUALITY=0
CLOUD_QUALITY=1
CLOUD_SHADOWS=true
DETAIL_QUALITY=0
DOUBLE_REIM_CLOUDS=true
FXAA_DEFINE=-1
HAND_SWAYING=1
LIGHTSHAFT_QUALI_DEFINE=0
MOON_PHASE_INF_ATMOSPHERE=true
MOON_PHASE_INF_LIGHT=true
SHADOW_QUALITY=-1
WATER_REFLECT_QUALITY=1
shadowDistance=64.0
```

**Comment by EminGT** [comment]
I think yo should first try updating your Iris and Sodium to the latest versions. If that doesn't fix it try with no mods except for those.

**Comment by Y4k3d0** [comment]
Will do. I understand this is much of an edge case. Thank you for your time !

**Comment by Y4k3d0** [comment]
Out of curiosity, I investigated further and managed to narrow the mod list down to Sodium, Iris and More Culling. I can't be 100% sure it's the culprit since I haven't been able to reproduce it w/ Sodium+Iris only but this may be a false negative. I am sure of one thing though, every time it occurred More Culling was active.

Also, in most cases I noticed the entity starts glowing when the crosshair's height aligns with its top border. It doesn't have to be perfectly on top of the frame but close enough. However when I logged back on the artifact was gone and didn't seem to appear if I stayed on the same spot.

In short, I can predict the camera angle (kind of) but its position looks a bit random to me for now.

**Comment by Y4k3d0** [comment]
It seems that some people were having some issues with the custom item frame renderer from More Culling, so this may be related to it. I'll try disabling it for now.

**Comment by EminGT** [comment]
Did you update Sodium and Iris to the latest versions? Did you saw the problem happening on the latest versions?

**Comment by Y4k3d0** [comment]
I tried on the latest version but I wasn't able to reproduce it.

**Comment by EminGT** [comment]
Assuming this is fixed with the latest Sodium and Iris then
